subsys/usb/device_next/class/usbd_cdc_acm.c

```diff
@@ -540,6 +544,11 @@ static void cdc_acm_tx_fifo_handler(struct k_work *work)
 		return;
 	}
 
+	if (atomic_test_and_set_bit(&data->state, CDC_ACM_TX_EP_BUSY)) {
+		cdc_acm_work_submit(&data->tx_fifo_work);
```

Comment:
How does this work? Will it simply keep retriggering the TX fifo work until the request is completed? This sounds way too much CPU intensive to me that could starve system workqueue and other lower priority threads.

Reply:
Hi @tmon-nordic, you are right. Indeed keep requeuing tx_fifo_work would cause the issue you mentioned, I have totally missed that. I have changed the handling to raised the tx_fifo altered flag instead. Please help and see whether it make sense for you.